### DISCUSSION TITLE
Distributed Akvorado

### DISCUSSION THREAD

**Discussion by knpshkn** [akvorado/akvorado#58] [Q&A]
First of all, I would like to thank you for this great product. you are doing a really great job
Secondly, i apologize in advance if I ask a stupid question :)

We would like to build a distributed system, for example, in 5 locations. In each location, for example, to have services: orchestrator, kafka, clickhouse, inlet, (zookeeper?).
The network equipment in each location will send sflow to the address of local inlet service of its location.
In the 5th location we plan to deploy console service, (traffic?) and distributed clickhouse, which will merge data from each location
Our goal is for each location (with inlet service) to be autonomous and these locations don't know anything about each other
We would like to ask you about the distributed system, how do you see it? Maybe you can give some advice to us

for example:
![sample](https://user-images.githubusercontent.com/49982467/181290560-7574a17f-4523-4fa3-9815-a35b53ab7a50.png)

**Comment by vincentbernat** [comment]
Out-of-the-box, I would distribute the inlet in each location, but you'll have a global Kafka cluster (distributed among the locations if you want) and a global ClickHouse server (at a central location). If some location becomes unavailable, this should continue to work. This way, you only have to distribute the Kafka cluster (which is easy).

Ultimately, the components are relatively independant. You can opt out the orchestrator as long as you configure the Kafka topic and the ClickHouse database yourself.

**Comment by fstolba** [comment]
Expanding on this, how would one go about configuring Akvorado to use TLS to encrypt communication between the Inlet and the Kafka broker?

**Reply by vincentbernat** [reply]
TLS has not been implemented, but that's not difficult to add (just some wiring with the underlying Go library). Open an issue about that if you want.
